server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -586,14 +594,9 @@ public void run() {
      * the metadata table and remove any migrating tablets that no longer exist.
      */
     private void cleanupNonexistentMigrations(final ClientContext clientContext) {
+      Map<DataLevel,Set<KeyExtent>> notSeen = partitionMigrations();
```

Comment:
This entire function can probably be removed.  The set of migrations in memory could get out of sync with the metadata table. With migrations stored in the metadata table this should not happen if the following conditions are met.

 1. Migrations are added using a conditional mutation that checks tablet exists
 2. Splits and merge operation on tablet remove migrations.
 3. Deleting a table delete all of its metadata including migrations.  This should happen w/o any changes in this PR.

These conditions are being met by the changes in this PR AFAICT.

Reply:
Removed, thanks for the added info. Wasn't sure if this method could be removed.

> Migrations are added using a conditional mutation that checks tablet exists

`tabletsMutator.mutateTablet(ke).putMigration(TabletServerIdImpl.toThrift(m.getNewTabletServer())).mutate();` is how migrations are added. Does this meet this condition

Reply:
> tabletsMutator.mutateTablet(ke).putMigration(TabletServerIdImpl.toThrift(m.getNewTabletServer())).mutate(); (see Manager.balanceTablets for more context) is how migrations are added. Does this meet this condition

That is an unconditional write to the metadata table.  When the write runs the tablet may no longer exists (tablet could have been merged away or the tablet delete) but the write will still happen.  That could lead to junk in the metadata table that will make other parts of accumulo unhappy.  Should use a conditional write there.